### ISSUE TITLE
Incorrect stderr redirection in Tooling API

### ISSUE THREAD

**Issue by blindpirate** [gradle/gradle#7617]
### Context

Since 4.7, seems like Tooling API redirect both standard error and standard output of target invocation to standard output of source invocation.

### Steps to Reproduce (for bugs)

Target project `build.gradle`:

```
System.out.println("This is stdout")
  
System.err.println("This is stderr")
```

Src project `build.gradle`:

```
import org.gradle.tooling.*
  
ProjectConnection connection = GradleConnector.newConnector().forProjectDirectory(<target project>).connect()

BuildLauncher build = connection.newBuild()

build.setStandardOutput(new FileOutputStream('stdout.txt'))
build.setStandardError(new FileOutputStream('stderr.txt'))

build.run()
```

Gradle 4.6's behavior is correct: `stderr.txt` contains `This is stderr`. In Gradle 4.7+ `stderr.txt` is empty, all outputs are redirected to `stdout.txt`.

According to `git bisect`, this is introduced by https://github.com/gradle/gradle/commit/71f5fe8e00ed5e1bedc0dddcdcdf1091cbebe92f

@adammurdoch Is this expected behavior?

**Comment by adammurdoch** [comment]
It is expected behaviour, or was at the time the change was made. We changed the behaviour when the plain console (intentionally) started writing all logging to stdout, just like the rich console does, but we did not change it back when the plain console started writing some logging to stderr again. 

We should change the behaviour so that it is consistent with what happens on the command-line (whatever that happens to be).
